src/iarc7_motion/iarc_tasks/height_holder.py

```diff
@@ -6,32 +6,40 @@
 
 import math
 import rospy
+import threading
 
 from geometry_msgs.msg import TwistStamped
 from nav_msgs.msg import Odometry
 
 class HeightHolder():
     def __init__(self):
-        self._k_z = .1 #p-term for z-height
         self._odometry = None
+        self._lock = threading.RLock()
         self._current_velocity_sub = rospy.Subscriber('/odometry/filtered',
                                               Odometry,
                                               self._current_velocity_callback)
         try:
+            self._MIN_MANEUVER_HEIGHT = rospy.get_param('~min_maneuver_height')
             self._TRACK_HEIGHT = rospy.get_param('~track_roomba_height')
+            self._K_Z = rospy.get_param('~k_term_hold_z')
         except KeyError as e:
             rospy.logerr('Could not lookup a parameter for track roomba task')
             raise
 
-        while self._odometry is None:
-            pass
+        if self._TRACK_HEIGHT < self._MIN_MANEUVER_HEIGHT:
+            raise ValueError('Track Roomba height was below the minimum maneuver height')
 
-    #uses a p-controller to return a velocity to maintain a height
-    #that is set as the param _track_roomba_height
+    # uses a p-controller to return a velocity to maintain a height
+    # that is set as the param _track_roomba_height
     def get_height_hold_response(self):
-        delta_z = self._TRACK_HEIGHT - self._odometry.pose.pose.position.z 
-        response = self._k_z * delta_z + self._odometry.twist.twist.linear.z
-        return response
+        with self._lock:
+            if self._odometry.pose.pose.position.z  < self._MIN_MANEUVER_HEIGHT:
+                raise ValueError('Drone height was below the minimum maneuver height')
```

Comment:
Don't need to raise a value error, instead you could return a TaskAborted for instanced.